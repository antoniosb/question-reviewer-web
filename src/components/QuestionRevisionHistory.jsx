import React from 'react';
import { Modal, Button, Row, Col } from 'antd'
import formatDateTime from '../utils/date'
import '../assets/scss/components/question-revision-history.scss'

class QuestionRevisionHistory extends React.PureComponent {
  constructor (props){
    super(props)
    let ultima = {}
    const question = this.props.value
    const revisions = (question.question_revisions || []).reverse()

    if (revisions.length > 0) {
      ultima = revisions[0]
    }
    this.state = { modalVisible: false, ultima, question, revisions }
  }

  render() {
    const { modalVisible, question, ultima, revisions } = this.state
    if (revisions.length === 0) {
      return (<div />)
    }
    return (
      <div className="question-revision-history">
        <Button icon="clock-circle-o" onClick={() => this.setState({ modalVisible: true })} />
        <h4>Última Revisão ({formatDateTime(ultima.updated_at)}): </h4> { ultima.comment }
        <Modal visible={modalVisible} title={`Revisões da Questão #${this.props.value.id}`} footer={[
            <Button onClick={() => this.setState({ modalVisible: false })}>Fechar</Button>
          ]}>
        { revisions.map((rev, idx) =>
          <div className="question-revision-history-item">
            <h4>{formatDateTime(rev.updated_at)}: </h4>{rev.comment}
          </div>
        )}
        </Modal>
      </div>
    )
  }
}

export default QuestionRevisionHistory
