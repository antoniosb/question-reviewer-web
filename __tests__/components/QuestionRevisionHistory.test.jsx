import React from 'react'
import QuestionRevisionHistory from '../../src/components/QuestionRevisionHistory'
import { Button, Modal } from 'antd';

describe('Components > QuestionRevisionHistory', () => {
  test('Render Nothing', () => {
    const component = shallow(<QuestionRevisionHistory value={{}} />)

    expect(component).toMatchSnapshot()
  })
  test('Render Nothing empty revisions', () => {
    const component = shallow(<QuestionRevisionHistory value={{ question_revisions: [] }} />)

    expect(component).toMatchSnapshot()
  })
  test('Render', () => {
    const component = shallow(<QuestionRevisionHistory value={{ question_revisions: [{ updated_at: new Date(2000, 1, 1, 12, 1, 1).toISOString(), comment: 'Test' }] }} />)
    expect(component).toMatchSnapshot()
    component.find(Button).simulate('click')
    expect(component).toMatchSnapshot()
    component.find(Modal).get(0).props.footer[0].props.onClick()
    expect(component).toMatchSnapshot()
  })
})