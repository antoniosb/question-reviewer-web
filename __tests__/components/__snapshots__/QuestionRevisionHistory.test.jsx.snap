// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > QuestionRevisionHistory Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionRevisionHistory
    value={
      Object {
        "question_revisions": Array [
          Object {
            "comment": "Test",
            "updated_at": "2000-02-01T14:01:01.000Z",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          ghost={false}
          icon="clock-circle-o"
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
        />,
        <h4>
          Última Revisão (
          01/02/2000 às 12:01
          ): 
        </h4>,
        " ",
        "Test",
        <Modal
          confirmLoading={false}
          footer={
            Array [
              <Button
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Fechar
              </Button>,
            ]
          }
          maskTransitionName="fade"
          okType="primary"
          prefixCls="ant-modal"
          title="Revisões da Questão #undefined"
          transitionName="zoom"
          visible={false}
          width={520}
        >
          <div
            className="question-revision-history-item"
          >
            <h4>
              01/02/2000 às 12:01
              : 
            </h4>
            Test
          </div>
        </Modal>,
      ],
      "className": "question-revision-history",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ghost": false,
          "icon": "clock-circle-o",
          "loading": false,
          "onClick": [Function],
          "prefixCls": "ant-btn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Última Revisão (",
            "01/02/2000 às 12:01",
            "): ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Última Revisão (",
          "01/02/2000 às 12:01",
          "): ",
        ],
        "type": "h4",
      },
      " ",
      "Test",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="question-revision-history-item"
            >
              <h4>
                01/02/2000 às 12:01
                : 
              </h4>
              Test
            </div>,
          ],
          "confirmLoading": false,
          "footer": Array [
            <Button
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            >
              Fechar
            </Button>,
          ],
          "maskTransitionName": "fade",
          "okType": "primary",
          "prefixCls": "ant-modal",
          "title": "Revisões da Questão #undefined",
          "transitionName": "zoom",
          "visible": false,
          "width": 520,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  01/02/2000 às 12:01
                  : 
                </h4>,
                "Test",
              ],
              "className": "question-revision-history-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "01/02/2000 às 12:01",
                    ": ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "01/02/2000 às 12:01",
                  ": ",
                ],
                "type": "h4",
              },
              "Test",
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            ghost={false}
            icon="clock-circle-o"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
          />,
          <h4>
            Última Revisão (
            01/02/2000 às 12:01
            ): 
          </h4>,
          " ",
          "Test",
          <Modal
            confirmLoading={false}
            footer={
              Array [
                <Button
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                >
                  Fechar
                </Button>,
              ]
            }
            maskTransitionName="fade"
            okType="primary"
            prefixCls="ant-modal"
            title="Revisões da Questão #undefined"
            transitionName="zoom"
            visible={false}
            width={520}
          >
            <div
              className="question-revision-history-item"
            >
              <h4>
                01/02/2000 às 12:01
                : 
              </h4>
              Test
            </div>
          </Modal>,
        ],
        "className": "question-revision-history",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ghost": false,
            "icon": "clock-circle-o",
            "loading": false,
            "onClick": [Function],
            "prefixCls": "ant-btn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Última Revisão (",
              "01/02/2000 às 12:01",
              "): ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Última Revisão (",
            "01/02/2000 às 12:01",
            "): ",
          ],
          "type": "h4",
        },
        " ",
        "Test",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="question-revision-history-item"
              >
                <h4>
                  01/02/2000 às 12:01
                  : 
                </h4>
                Test
              </div>,
            ],
            "confirmLoading": false,
            "footer": Array [
              <Button
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Fechar
              </Button>,
            ],
            "maskTransitionName": "fade",
            "okType": "primary",
            "prefixCls": "ant-modal",
            "title": "Revisões da Questão #undefined",
            "transitionName": "zoom",
            "visible": false,
            "width": 520,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    01/02/2000 às 12:01
                    : 
                  </h4>,
                  "Test",
                ],
                "className": "question-revision-history-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "01/02/2000 às 12:01",
                      ": ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "01/02/2000 às 12:01",
                    ": ",
                  ],
                  "type": "h4",
                },
                "Test",
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Components > QuestionRevisionHistory Render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionRevisionHistory
    value={
      Object {
        "question_revisions": Array [
          Object {
            "comment": "Test",
            "updated_at": "2000-02-01T14:01:01.000Z",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          ghost={false}
          icon="clock-circle-o"
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
        />,
        <h4>
          Última Revisão (
          01/02/2000 às 12:01
          ): 
        </h4>,
        " ",
        "Test",
        <Modal
          confirmLoading={false}
          footer={
            Array [
              <Button
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Fechar
              </Button>,
            ]
          }
          maskTransitionName="fade"
          okType="primary"
          prefixCls="ant-modal"
          title="Revisões da Questão #undefined"
          transitionName="zoom"
          visible={true}
          width={520}
        >
          <div
            className="question-revision-history-item"
          >
            <h4>
              01/02/2000 às 12:01
              : 
            </h4>
            Test
          </div>
        </Modal>,
      ],
      "className": "question-revision-history",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ghost": false,
          "icon": "clock-circle-o",
          "loading": false,
          "onClick": [Function],
          "prefixCls": "ant-btn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Última Revisão (",
            "01/02/2000 às 12:01",
            "): ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Última Revisão (",
          "01/02/2000 às 12:01",
          "): ",
        ],
        "type": "h4",
      },
      " ",
      "Test",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="question-revision-history-item"
            >
              <h4>
                01/02/2000 às 12:01
                : 
              </h4>
              Test
            </div>,
          ],
          "confirmLoading": false,
          "footer": Array [
            <Button
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            >
              Fechar
            </Button>,
          ],
          "maskTransitionName": "fade",
          "okType": "primary",
          "prefixCls": "ant-modal",
          "title": "Revisões da Questão #undefined",
          "transitionName": "zoom",
          "visible": true,
          "width": 520,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  01/02/2000 às 12:01
                  : 
                </h4>,
                "Test",
              ],
              "className": "question-revision-history-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "01/02/2000 às 12:01",
                    ": ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "01/02/2000 às 12:01",
                  ": ",
                ],
                "type": "h4",
              },
              "Test",
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            ghost={false}
            icon="clock-circle-o"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
          />,
          <h4>
            Última Revisão (
            01/02/2000 às 12:01
            ): 
          </h4>,
          " ",
          "Test",
          <Modal
            confirmLoading={false}
            footer={
              Array [
                <Button
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                >
                  Fechar
                </Button>,
              ]
            }
            maskTransitionName="fade"
            okType="primary"
            prefixCls="ant-modal"
            title="Revisões da Questão #undefined"
            transitionName="zoom"
            visible={true}
            width={520}
          >
            <div
              className="question-revision-history-item"
            >
              <h4>
                01/02/2000 às 12:01
                : 
              </h4>
              Test
            </div>
          </Modal>,
        ],
        "className": "question-revision-history",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ghost": false,
            "icon": "clock-circle-o",
            "loading": false,
            "onClick": [Function],
            "prefixCls": "ant-btn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Última Revisão (",
              "01/02/2000 às 12:01",
              "): ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Última Revisão (",
            "01/02/2000 às 12:01",
            "): ",
          ],
          "type": "h4",
        },
        " ",
        "Test",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="question-revision-history-item"
              >
                <h4>
                  01/02/2000 às 12:01
                  : 
                </h4>
                Test
              </div>,
            ],
            "confirmLoading": false,
            "footer": Array [
              <Button
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Fechar
              </Button>,
            ],
            "maskTransitionName": "fade",
            "okType": "primary",
            "prefixCls": "ant-modal",
            "title": "Revisões da Questão #undefined",
            "transitionName": "zoom",
            "visible": true,
            "width": 520,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    01/02/2000 às 12:01
                    : 
                  </h4>,
                  "Test",
                ],
                "className": "question-revision-history-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "01/02/2000 às 12:01",
                      ": ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "01/02/2000 às 12:01",
                    ": ",
                  ],
                  "type": "h4",
                },
                "Test",
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Components > QuestionRevisionHistory Render 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionRevisionHistory
    value={
      Object {
        "question_revisions": Array [
          Object {
            "comment": "Test",
            "updated_at": "2000-02-01T14:01:01.000Z",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          ghost={false}
          icon="clock-circle-o"
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
        />,
        <h4>
          Última Revisão (
          01/02/2000 às 12:01
          ): 
        </h4>,
        " ",
        "Test",
        <Modal
          confirmLoading={false}
          footer={
            Array [
              <Button
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Fechar
              </Button>,
            ]
          }
          maskTransitionName="fade"
          okType="primary"
          prefixCls="ant-modal"
          title="Revisões da Questão #undefined"
          transitionName="zoom"
          visible={true}
          width={520}
        >
          <div
            className="question-revision-history-item"
          >
            <h4>
              01/02/2000 às 12:01
              : 
            </h4>
            Test
          </div>
        </Modal>,
      ],
      "className": "question-revision-history",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ghost": false,
          "icon": "clock-circle-o",
          "loading": false,
          "onClick": [Function],
          "prefixCls": "ant-btn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Última Revisão (",
            "01/02/2000 às 12:01",
            "): ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Última Revisão (",
          "01/02/2000 às 12:01",
          "): ",
        ],
        "type": "h4",
      },
      " ",
      "Test",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="question-revision-history-item"
            >
              <h4>
                01/02/2000 às 12:01
                : 
              </h4>
              Test
            </div>,
          ],
          "confirmLoading": false,
          "footer": Array [
            <Button
              ghost={false}
              loading={false}
              onClick={[Function]}
              prefixCls="ant-btn"
            >
              Fechar
            </Button>,
          ],
          "maskTransitionName": "fade",
          "okType": "primary",
          "prefixCls": "ant-modal",
          "title": "Revisões da Questão #undefined",
          "transitionName": "zoom",
          "visible": true,
          "width": 520,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  01/02/2000 às 12:01
                  : 
                </h4>,
                "Test",
              ],
              "className": "question-revision-history-item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "01/02/2000 às 12:01",
                    ": ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "01/02/2000 às 12:01",
                  ": ",
                ],
                "type": "h4",
              },
              "Test",
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            ghost={false}
            icon="clock-circle-o"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
          />,
          <h4>
            Última Revisão (
            01/02/2000 às 12:01
            ): 
          </h4>,
          " ",
          "Test",
          <Modal
            confirmLoading={false}
            footer={
              Array [
                <Button
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                >
                  Fechar
                </Button>,
              ]
            }
            maskTransitionName="fade"
            okType="primary"
            prefixCls="ant-modal"
            title="Revisões da Questão #undefined"
            transitionName="zoom"
            visible={true}
            width={520}
          >
            <div
              className="question-revision-history-item"
            >
              <h4>
                01/02/2000 às 12:01
                : 
              </h4>
              Test
            </div>
          </Modal>,
        ],
        "className": "question-revision-history",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ghost": false,
            "icon": "clock-circle-o",
            "loading": false,
            "onClick": [Function],
            "prefixCls": "ant-btn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Última Revisão (",
              "01/02/2000 às 12:01",
              "): ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Última Revisão (",
            "01/02/2000 às 12:01",
            "): ",
          ],
          "type": "h4",
        },
        " ",
        "Test",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="question-revision-history-item"
              >
                <h4>
                  01/02/2000 às 12:01
                  : 
                </h4>
                Test
              </div>,
            ],
            "confirmLoading": false,
            "footer": Array [
              <Button
                ghost={false}
                loading={false}
                onClick={[Function]}
                prefixCls="ant-btn"
              >
                Fechar
              </Button>,
            ],
            "maskTransitionName": "fade",
            "okType": "primary",
            "prefixCls": "ant-modal",
            "title": "Revisões da Questão #undefined",
            "transitionName": "zoom",
            "visible": true,
            "width": 520,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    01/02/2000 às 12:01
                    : 
                  </h4>,
                  "Test",
                ],
                "className": "question-revision-history-item",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "01/02/2000 às 12:01",
                      ": ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "01/02/2000 às 12:01",
                    ": ",
                  ],
                  "type": "h4",
                },
                "Test",
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Components > QuestionRevisionHistory Render Nothing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionRevisionHistory
    value={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Components > QuestionRevisionHistory Render Nothing empty revisions 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionRevisionHistory
    value={
      Object {
        "question_revisions": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
